src/traversals/bfs.jl

```diff
@@ -265,3 +234,49 @@ function bipartite_map(g::SimpleGraph)
     end
     m
 end
+
+###########################################
+# Get the map of the (geodesic) distances from vertices to source #
+###########################################
+
+"""
+    gdistances!(g, source, dists) -> dists
+
+Fills `dists` with the geodesic distances of vertices in `g` from vertex/vertices `source`.
+`dists` should be a vector of length `nv(g)`.
+"""
+function gdistances!(g::SimpleGraph, source, dists)
+    n = nv(g)
+    fill!(dists, -1)
+    queue = Vector{Int}(n)
+    for i in 1:length(source)
+        queue[i] = source[i]
+        dists[source[i]] = 0
+    end
+    head = 1
+    tail = length(source)
+    while head <= tail
```

Comment:
This is a bespoke implementation of BFS can it be handled within the Visitor Framework?